packages/web/src/pages/upload-page/fields/StemFilesView.tsx

```diff
@@ -31,25 +33,83 @@ const messages = {
   stemTypeDescription: 'Please select a stem type for each of your files.'
 }
 
+const useStemFileInfos = (stems: StemUploadWithFile[]) => {
+  const [fileInfos, setFileInfos] = useState<{ [index: number]: File }>({})
+
+  useEffect(() => {
+    const indexToTrackIdsMap = stems.reduce((acc, stem, i) => {
+      if (!stem.file) acc[i] = stem.metadata.track_id
+      return acc
+    }, {})
+    const indexToTrackTitlesMap = stems.reduce((acc, stem, i) => {
+      if (!stem.file) acc[i] = stem.metadata.orig_filename ?? ''
+      return acc
+    }, {})
+
+    const fetchInfos = async (indexToTrackIdsMap: {
+      [index: number]: number
+    }) => {
+      try {
+        const sdk = await audiusSdk()
+        const indices = Object.keys(indexToTrackIdsMap) as unknown as number[]
```

Comment:
why is the `as unknown as number[]` necessary? aren't the keys of `indexToTrackIdsMap` already numbers?

Reply:
Object.keys returns strings. going to use .map(Number), thx